lib/jwt_signed_request/errors.rb

```diff
@@ -0,0 +1,12 @@
+module JWTSignedRequest
+  UnauthorizedRequestError = Class.new(StandardError)
+  MissingAuthorizationHeaderError = Class.new(UnauthorizedRequestError)
+  JWTDecodeError = Class.new(UnauthorizedRequestError)
+  RequestVerificationFailedError = Class.new(UnauthorizedRequestError)
+  %w[Method Path Header Body Query].each do |type|
+    module_eval("Request#{type}VerificationFailedError = Class.new(RequestVerificationFailedError)")
+  end
```

Comment:
Somewhat personal preference, but I dont think the 2 line saving is worth the loss in grepability here vs just listing these explicitly